docs/_source/tutorials/notebooks/training-llm-mistral-sft.ipynb

```diff
@@ -0,0 +1,602 @@
+       "<datasets.iterable_dataset.IterableDataset at 0x22fa12c73d0>"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from datasets import load_dataset\n",
+    "\n",
+    "data = load_dataset(\"laion/OIG\", split=\"train\", streaming=True)\n",
+    "# Randomly sample through shuffle + take, but note that shuffling with a streaming\n",
+    "# dataset works with a buffer size, so the random 30k samples are not truly random\n",
+    "# Reduce the buffer size to speed up the data selection process.\n",
+    "data = data.shuffle(buffer_size=1_000_000).take(30_000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The loaded data has a \"text\" column with data that looks like so:\n",
+    "```\n",
+    "Background: Tubulin in molecular biology can refer either to the tubulin protein superfamily of globular proteins, or one of the member proteins of that superfamily. α- and β-tubulins polymerize into microtubules, a major component of the eukaryotic cytoskeleton. Microtubules function in many essential cellular processes, including mitosis. Tubulin-binding drugs kill cancerous cells by inhibiting microtubule dynamics, which are required for DNA segregation and therefore cell division. , there are six members of the tubulin superfamily, although not all are present in all species. Both α and β tubulins have a mass of around 50 kDa and are thus in a similar range compared to actin (with a mass of ~42 kDa). In contrast, tubulin polymers (microtubules) tend to be much bigger than actin filaments due to their cylindrical nature. Tubulin was long thought to be specific to eukaryotes. More recently, however, several prokaryotic proteins have been shown to be related to tubulin. <human>: What are some examples of drugs that bind to tubulin? <bot>: These include paclitaxel, colchicine, and the \"vinca\" alkaloids, each of which have a distinct binding site on β-tubulin. <human>: Are there any other examples? <bot>: In addition, the anti-worm drug albendazole selectively targets the colchicine site in worm, but not human or \"Drosophilia\", β-Tubulin. <human>: What is another specific example of a drug that binds to tubulin? <bot>: Class III β-tubulin is a microtubule element expressed exclusively in neurons, and is a popular identifier specific for neurons in nervous tissue. <human>: What is the difference between class III and other isotypes? <bot>: It binds colchicine much more slowly than other isotypes of β-tubulin. <human>: Are there six classes of isotypes? <bot>: β1-tubulin, sometimes called class VI β-tubulin, is the most divergent at the amino acid sequence level. <human>: What are the functions of the different isotypes? <bot>: It is expressed exclusively in megakaryocytes and platelets in humans and appears to play an important role in the formation of platelets.\n",
+    "```\n",
+    "Let's break this down:\n",
```

Comment:
maybe mention that this differs per model?

Reply:
It doesn't, this is just specific to the data, not the model.

And for the model we make our own template, which you can always do when finetuning a base model.

Reply:
Yes, I think it is a two-fold thing which also includes configuring model token format later on in the process. I think it might be nice to briefly mention the complexities/background of this, similar to the reason why they added the tokenizer implementation to transformers right?